src/rules/ResourceTargetChecker.js

```diff
@@ -45,107 +46,40 @@ class ResourceTargetChecker extends Rule {
      */
     constructor(options) {
         super(options);
-
-        if (!options || !options.name || !options.description || !options.note || !options.regexps) {
-            throw "Missing required options for the ResourceMatcher constructor";
-        }
-        ["name", "description", "regexps", "note", "sourceLocale", "link"].forEach(prop => {
-            this[prop] = options[prop];
-        });
-        this.sourceLocale = this.sourceLocale || "en-US";
-
-        // this may throw if you got to the regexp syntax wrong:
-        this.re = this.regexps.map(regexp => new RegExp(regexp, "gu"));
-    }
-
-    getRuleType() {
-        return "resource";
     }
 
     /**
      * @override
      */
-    match(options) {
-        const { locale, resource, file } = options || {};
-        const _this = this;
-
-        /**
-         * @private
-         */
-        function checkString(re, src, tar) {
-            re.lastIndex = 0;
-            let matches = [];
-            const strippedTar = stripPlurals(tar);
-
-            // check the target only
-            re.lastIndex = 0;
-            let match = re.exec(strippedTar);
-            while (match) {
-                let value = {
-                    severity: "error",
-                    id: resource.getKey(),
-                    source: src,
-                    rule: _this,
-                    pathName: file,
-                    highlight: `Target: ${tar.substring(0, match.index)}<e0>${match[0]}</e0>${tar.substring(match.index+match[0].length)}`,
-                    description: _this.note.replace(/\{matchString\}/g, match[0])
-                };
-                if (typeof(options.lineNumber) !== 'undefined') {
-                    value.lineNumber = options.lineNumber;
-                }
-                matches.push(new Result(value));
-
-                match = re.exec(strippedTar);
+    checkString(re, src, tar, file, resource) {
+        re.lastIndex = 0;
+        let matches = [];
+        const strippedTar = stripPlurals(tar);
+
+        // check the target only, but we need the source in order
+        // to construct a Result if necessary
+        re.lastIndex = 0;
+        let match = re.exec(strippedTar);
+        while (match) {
+            let value = {
+                severity: this.severity,
+                id: resource.getKey(),
+                source: src,
+                rule: this,
+                pathName: file,
+                highlight: `Target: ${tar.substring(0, match.index)}<e0>${match[0]}</e0>${tar.substring(match.index+match[0].length)}`,
+                description: this.note.replace(/\{matchString\}/g, match[0])
```

Comment:
Missing locale. Suggestions: we could either pass the locale down to checkString, or have checkString return partial result (object) instead, which gets assembled into a proper Result within DeclarativeResourceRule's matchString

Reply:
Actually, resources are special in that they contain their own locales, both source and target. So, I think it would be better to just remove the locale completely from the parameters and use the locale in the resource itself. I've updated the code to create the Result instances with the appropriate locale from the resource.